platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -50,15 +50,24 @@ static mender_err_t
 nvs_read_alloc(struct nvs_fs *nvs, uint16_t id, void **data, size_t *length) {
     ssize_t ret;
 
-    /* Retrieve length of the data */
-    ret = nvs_read(nvs, id, NULL, 0);
+    /* Allocate 1 byte fro the first peek read */
+    *data = malloc(1);
+    if (NULL == *data) {
+        mender_log_error("Unable to allocate memory for: %d", id);
+        return MENDER_FAIL;
+    }
+
+    /* Peek read to retrieve length of the data */
+    ret = nvs_read(nvs, id, *data, 0);
```

Comment:
I think it's better to do `unsigned char byte` and then `nvs_read(nvs, id, &byte, 0)`; You don't need heap-allocated space to read into, a stack-allocated variable is good enough and way cheaper.

Reply:
Of course, thank you.